File: resources/vue/src/api/es-index.js
import request from '@/utils/request'

const api = '/api/es_index/'

export function CreateAction(data) {
  return request({
    url: api + 'CreateAction',
    method: 'post',
    data
  })
}

export function GetSettingsAction(data) {
  return request({
    url: api + 'GetSettingsAction',
    method: 'post',
    data
  })
}

export function IndexNamesAction(data) {
  return request({
    url: api + 'IndexNamesAction',
    method: 'post',
    data
  })
}

export function ReindexAction(data) {
  return request({
    url: api + 'ReindexAction',
    method: 'post',
    data
  })
}

export function GetAliasAction(data) {
  return request({
    url: api + 'GetAliasAction',
    method: 'post',
    data
  })
}

export function DeleteAction(data) {
  return request({
    url: api + 'DeleteAction',
    method: 'post',
    data
  })
}

export function MoveAliasToIndex(data) {
  return request({
    url: api + 'MoveAliasToIndex',
    method: 'post',
    data
  })
}

export function AddAliasToIndex(data) {
  return request({
    url: api + 'AddAliasToIndex',
    method: 'post',
    data
  })
}
export function BatchAddAliasToIndex(data) {
  return request({
    url: api + 'BatchAddAliasToIndex',
    method: 'post',
    data
  })
}

export function RemoveAlias(data) {
  return request({
    url: api + 'RemoveAlias',
    method: 'post',
    data
  })
}

export function GetSettingsInfoAction(data) {
  return request({
    url: api + 'GetSettingsInfoAction',
    method: 'post',
    data
  })
}

export function StatsAction(data) {
  return request({
    url: api + 'StatsAction',
    method: 'post',
    data
  })
}

export function CatStatusAction(data) {
  return request({
    url: api + 'CatStatusAction',
    method: 'post',
    data
  })
}

export function IndexsCountAction(data) {
  return request({
    url: api + 'IndexsCountAction',
    method: 'post',
    data
  })
}

File: pkg/web/es_index.go
package web

// ES索引 路由
func (this *WebServer) runEsIndex() {

	const AbsolutePath = "/api/es_index"
	group := this.engine.Group("ES索引", AbsolutePath)
	{
		group.POST(false, "删除ES索引", "/DeleteAction", this.esIndexController.DeleteAction)
		group.POST(false, "创建ES索引", "/CreateAction", this.esIndexController.CreateAction)
		group.POST(false, "获取ES索引配置", "/GetSettingsAction", this.esIndexController.GetSettingsAction)
		group.POST(false, "得到所有的ES索引名", "/IndexNamesAction", this.esIndexController.IndexNamesAction)
		group.POST(false, "重建索引", "/ReindexAction", this.esIndexController.ReindexAction)
		group.POST(false, "获取别名", "/GetAliasAction", this.esIndexController.GetAliasAction)
		group.POST(false, "迁移别名到其他索引", "/MoveAliasToIndex", this.esIndexController.MoveAliasToIndex)
		group.POST(false, "添加别名", "/AddAliasToIndex", this.esIndexController.AddAliasToIndex)
		group.POST(false, "批量添加别名", "/BatchAddAliasToIndex", this.esIndexController.BatchAddAliasToIndex)
		group.POST(false, "移除别名", "/RemoveAlias", this.esIndexController.RemoveAlias)

		group.POST(false, "获取所有的索引配置信息", "/GetSettingsInfoAction", this.esIndexController.GetSettingsInfoAction)
		group.POST(false, "获取索引的Stats", "/StatsAction", this.esIndexController.StatsAction)

		group.POST(false, "得到所有的索引数量", "/IndexsCountAction", this.esIndexController.IndexsCountAction)

	}
}
